platform/ios/src/MGLMapView.h

```diff
@@ -427,7 +427,7 @@ MGL_EXPORT IB_DESIGNABLE
     position within the map view. If `NO`, the user location annotation
     instantaneously moves to its new position.
  */
-- (void)setUserLocationVerticalAlignment:(MGLAnnotationVerticalAlignment)alignment animated:(BOOL)animated;
+- (void)setUserLocationVerticalAlignment:(MGLAnnotationVerticalAlignment)alignment animated:(BOOL)animated __attribute__((deprecated("Use -[MGLMapViewDelegate mapViewUserLocationAnchorPoint:] instead.")));;
```

Comment:
What’s the verdict on a method for updating the offset spontaneously? To avoid exposing a method that conflicts with the new delegate method, maybe we should just expose `-updateUserLocationAnnotation` publicly; calling that method would cause the new delegate method to be called automatically.

/ref https://github.com/mapbox/mapbox-gl-native/pull/12907#discussion_r219008704

Reply:
I think it is a good idea to allow the developer to update the offset spontaneously. I'll see what I can accomplish!